core/src/main/java/org/elasticsearch/env/NodeEnvironment.java

```diff
@@ -313,6 +315,20 @@ private void maybeLogHeapDetails() {
         logger.info("heap size [{}], compressed ordinary object pointers [{}]", maxHeapSize, useCompressedOops);
     }
 
+    private void maybeWarnFileDescriptors() {
+        long maxFileDescriptorCount = ProcessProbe.getInstance().getMaxFileDescriptorCount();
+        if (maxFileDescriptorCount == -1) {
+            return;
+        }
+        int fileDescriptorCountThreshold = (1 << 16);
+        if (maxFileDescriptorCount < fileDescriptorCountThreshold) {
+            logger.warn(
+                    "max file descriptors [{}] for elasticsearch process likely too low, consider increasing to [{}]",
```

Comment:
maybe use stronger wording? "consider increasing to at least [{}] (more is better)".